cnd/src/http_api/herc20_hbit/alice.rs

```diff
@@ -0,0 +1,402 @@
+                    hbit::FinalizedAsRedeemer {
+                        state: bitcoin_state,
+                        ..
+                    },
+                ..
+            } => Some(From::<hbit::State>::from(bitcoin_state.clone())),
+            _ => None,
+        }
+    }
+}
+
+impl InitAction
+    for AliceSwap<asset::Erc20, asset::Bitcoin, herc20::Finalized, hbit::FinalizedAsRedeemer>
+{
+    type Output = Never;
+    fn init_action(&self) -> anyhow::Result<Self::Output> {
+        anyhow::bail!(ActionNotFound)
+    }
+}
+
+impl From<AliceSwap<asset::Erc20, asset::Bitcoin, herc20::Finalized, hbit::FinalizedAsRedeemer>>
+    for Herc20
+{
+    fn from(
+        from: AliceSwap<asset::Erc20, asset::Bitcoin, herc20::Finalized, hbit::FinalizedAsRedeemer>,
+    ) -> Self {
+        match from {
+            AliceSwap::Created {
+                alpha_created: herc20_asset,
+                ..
+            }
+            | AliceSwap::Finalized {
+                alpha_finalized:
+                    herc20::Finalized {
+                        asset: herc20_asset,
+                        ..
+                    },
+                ..
+            } => Self {
+                protocol: "herc20".to_owned(),
+                quantity: herc20_asset.quantity.to_wei_dec(),
+                token_contract: herc20_asset.token_contract.to_string(),
+            },
+        }
+    }
+}
+
+impl From<AliceSwap<asset::Erc20, asset::Bitcoin, herc20::Finalized, hbit::FinalizedAsRedeemer>>
+    for Hbit
+{
+    fn from(
+        from: AliceSwap<asset::Erc20, asset::Bitcoin, herc20::Finalized, hbit::FinalizedAsRedeemer>,
+    ) -> Self {
+        match from {
+            AliceSwap::Created {
+                beta_created: bitcoin_asset,
+                ..
+            }
+            | AliceSwap::Finalized {
+                beta_finalized:
+                    hbit::FinalizedAsRedeemer {
+                        asset: bitcoin_asset,
+                        ..
+                    },
+                ..
+            } => Self {
+                protocol: "hbit".to_owned(),
+                quantity: bitcoin_asset.as_sat().to_string(),
+            },
+        }
+    }
+}
+
+impl AlphaLedger
+    for AliceSwap<asset::Erc20, asset::Bitcoin, herc20::Finalized, hbit::FinalizedAsRedeemer>
+{
+    fn alpha_ledger(&self) -> Ledger {
+        Ledger::Ethereum
+    }
+}
+
+impl BetaLedger
+    for AliceSwap<asset::Erc20, asset::Bitcoin, herc20::Finalized, hbit::FinalizedAsRedeemer>
+{
+    fn beta_ledger(&self) -> Ledger {
+        Ledger::Bitcoin
+    }
+}
+
+impl AlphaAbsoluteExpiry
+    for AliceSwap<asset::Erc20, asset::Bitcoin, herc20::Finalized, hbit::FinalizedAsRedeemer>
+{
+    fn alpha_absolute_expiry(&self) -> Option<Timestamp> {
+        match self {
+            AliceSwap::<
+                asset::Erc20,
+                asset::Bitcoin,
+                herc20::Finalized,
+                hbit::FinalizedAsRedeemer,
+            >::Created {
```

Comment:
I think there is quite some low-hanging fruit here :)

```suggestion
            AliceSwap::Created {
```

Reply:
ahhhhhhhhhhhhhhhhhhhhhhhh, how does that keep coming back?